nursereg/tasks.py

```diff
@@ -344,21 +339,27 @@ def update_create_vumi_contact(nursereg_id, client=None, sender=None):
                 # Do nothing if the cmsisdn has an active subscription
                 return contact
             else:
-                rmsisdn_active_subs = Subscription.objects.filter(
-                    to_addr=nursereg.rmsisdn, active=True,
-                    message_set__short_name="nurseconnect").order_by(
-                    'created_at')
-                if rmsisdn_active_subs.count() > 0:
-                    subscription = transfer_subscription(contact,
-                                                         rmsisdn_active_subs)
-                else:
+                try:
+                    # Get the old contact active subscription
+                    rmsisdn_active_sub = Subscription.objects.get(
+                        to_addr=nursereg.rmsisdn, active=True,
+                        message_set__short_name="nurseconnect")
+                    subscription = transfer_subscription(
+                        contact, rmsisdn_active_sub)
+                    # Update the contact with subscription details
+                    updated_contact = update_contact_subscription(
+                        contact, subscription, client)
+                    return updated_contact
+                except ObjectDoesNotExist:
                     # Create new subscription for the contact
                     subscription = create_subscription(contact, sender)
-
-                # Update the contact with subscription details
-                updated_contact = update_contact_subscription(
-                    contact, subscription, client)
-                return updated_contact
+                    # Update the contact with subscription details
+                    updated_contact = update_contact_subscription(
+                        contact, subscription, client)
+                    return updated_contact
+                except MultipleObjectsReturned:
```

Comment:
I think we're better off that an exception thrown into Sentry if there's multiple objects because it signals a failure down-stream. Remove it and add something like this for the test: `self.assertRaises(MultipleObjectsReturned,  create_contact()`